src/backend/distributed/commands/function.c

```diff
@@ -157,7 +164,11 @@ create_distributed_function(PG_FUNCTION_ARGS)
 	EnsureDependenciesExistsOnAllNodes(&functionAddress);
 
 	ddlCommand = GetFunctionDDLCommand(funcOid);
-	SendCommandToWorkers(ALL_WORKERS, ddlCommand);
+	foreach(ddlCommandCell, ddlCommand)
+	{
+		char *ddlCommandString = lfirst(ddlCommandCell);
+		SendCommandToWorkers(ALL_WORKERS, ddlCommandString);
```

Comment:
We should probably create this as the current user to ensure transactional semantics when running as non-superuser. We should also check ownership (or superuser) in this function before doing anything else.